index.bs

```diff
@@ -691,21 +691,124 @@ An {{MLOperand}} represents an intermediary graph being constructed as a result
+    1. Let |desc| be the fifth argument.
+    1. If |desc| is not an {{MLOperandDescriptor}}, throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+    1. If |type| is not equal to |desc|'s {{MLOperandDescriptor/type}}, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. Set [=this=]'s {{MLOperand/[[dimensions]]}} to |desc|.{{MLOperandDescriptor/dimensions}}.
+    1. Let |bufferView| be the sixth argument.
+    1. If |bufferView| is not an {{MLBufferView}}, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. If |type| is not equal to |bufferView|'s [=element type=], throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. If the [=byte length=] of |desc| is not supported by the implementation, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. If the [=byte length=] of |desc| is not equal with |bufferView|.\[[ByteLength]], throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. Set [=this=]'s {{MLOperand/[[tensorData]]}} to |bufferView|.
+1. Otherwise, of |kind| is `"input"`:
+    1. Let |name| be the fourth argument.
+    1. If |name| is not a [=string=], throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+    1. Set [=this=]'s {{MLOperand/[[name]]}} to |name|.
+    1. Let |desc| be the fifth argument.
+    1. If |desc| is not an {{MLOperandDescriptor}}, throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+    1. If |type| is not equal to |desc|'s {{MLOperandDescriptor/type}}, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. If the [=byte length=] of |desc| is not supported by the implementation, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. Set [=this=]'s {{MLOperand/[[dimensions]]}} to |desc|.{{MLOperandDescriptor/dimensions}}.
+1. Otherwise, of |kind| is `"output"`:
+    1. Let |dimensions| be the fourth argument.
+    1. If the [=check dimensions=] steps given |type| and |dimensions| return `false`, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. Let |activation| be the fifth argument.
+    1. If |activation| is not `undefined` and is not a [=string=] or a [=function=], throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. If |activation| is a [=string=] and is not supported by the implementation as an activation function, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. Make a request to the underlying platform to register |activation| as an activation function for [=this=].
+1. Return [=this=].
+
+To <dfn>check dimensions</dfn> given |dimensions| and |type|, run the following steps:
+1. If |dimensions| is not an array of positive numbers, return `false`;
+1. If |dimensions|.length is 0, return `false`.
+1. If |dimensions|.length is too large to be supported by the implementation, return `false`.
+1. If any element of |dimensions| is too large to be supported by the implementation, return `false`.
```

Comment:
Should the data type of element be validated, to be unsigned long? The value 0 (0 size dimension) should also be invalid.

Reply:
I don't know, I guess all integer type values should be accepted, the question is range.

Reply:
I think that may be implementation dependent. What we can explicitly check is that it's a positive ulong, but the implementation may have other constraints that it should be able to tell the API user if the specified dimension is not supported for whatever reason.